File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
import java.io.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        
        //오른쪽 괄호부터 시작하면 false
        if(s.charAt(0) == ')') return false;
        try{
            for(char c: s.toCharArray()){
                if(c == '('){
                    stack.push('c');
                }
                else if(c == ')'){
                    stack.pop();
                }
            }
        }
        catch(Exception e){
            return false;
        }
        
        return stack.size() == 0? true: false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    total = 0
    if s[0] == ')':
        return False
    
    for i in s:
        if i == '(':
            total += 1
        if i == ')':
            total += -1
        if total < 0:
            return False
        
    if total == 0:
        return True
    else:
        return False